src_c/polygon.c

```diff
@@ -305,6 +305,131 @@ pg_polygon_get_vertices(pgPolygonObject *self, void *closure)
+pg_polygon_ass_vertex(pgPolygonObject *self, Py_ssize_t i, PyObject *v)
+{
+    pgPolygonBase *poly = &self->polygon;
+
+    if (i < 0) {
+        i += poly->verts_num;
+    }
+
+    if (i < 0 || i > poly->verts_num) {
+        PyErr_SetString(PyExc_IndexError, "Invalid vertex Index");
+        return -1;
+    }
+
+    double v1, v2;
+    if (!pg_TwoDoublesFromObj(v, &v1, &v2)) {
+        PyErr_SetString(PyExc_TypeError, "Must assign numeric values");
+        return -1;
+    }
+    poly->vertices[i * 2] = v1;
+    poly->vertices[i * 2 + 1] = v2;
+    return 0;
+}
+
+static Py_ssize_t
+pg_polygon_seq_length(pgPolygonObject *self)
+{
+    pgPolygonBase *poly = &self->polygon;
+    return poly->verts_num;
+}
+
+static int
+pg_polygon_ass_subscript(pgPolygonObject *self, PyObject *op, PyObject *value)
+{
+    if (PyIndex_Check(op)) {
+        PyObject *index;
+        Py_ssize_t i;
+
+        index = PyNumber_Index(op);
+        if (index == NULL) {
+            return -1;
+        }
+        i = PyNumber_AsSsize_t(index, NULL);
+        Py_DECREF(index);
+        return pg_polygon_ass_vertex(self, i, value);
+    }
+    else {
+        PyErr_SetString(PyExc_TypeError, "Expected a number or sequence");
+        return -1;
+    }
+}
+
+static PyObject *
+pg_polygon_subscript(pgPolygonObject *self, PyObject *op)
+{
+    PyObject *vertices = _pg_polygon_vertices_aslist(&self->polygon);
+    PyObject *index = PyNumber_Index(op);
+    Py_ssize_t i;
+
+    if (index == NULL) {
+        return NULL;
+    }
+    i = PyNumber_AsSsize_t(index, NULL);
+    Py_DECREF(index);
+
+    pgPolygonBase *poly = &self->polygon;
+
+    if (i < 0) {
+        i += poly->verts_num;
+    }
+
+    if (i < 0 || i > poly->verts_num) {
+        return RAISE(PyExc_IndexError, "Invalid vertex Index");
+    }
+
+    return PyList_GetItem(vertices, i);
+}
+
+static int
+pg_polygon_contains_seq(pgPolygonObject *self, PyObject *arg)
+{
+    pgPolygonBase *poly = &self->polygon;
+
+    if (PyTuple_Check(arg) || PyList_Check(arg)) {
+        arg = PySequence_Tuple(arg);
+        Py_ssize_t i;
+        for (i = 0; i < poly->verts_num; i++) {
+            PyObject *tup = pg_TupleFromDoublePair(
+                poly->vertices[i * 2], poly->vertices[i * 2 + 1]);
+
+            double x1, y1, x2, y2;
+            if (!pg_DoubleFromObj(PyTuple_GetItem(tup, 0), &x1) ||
+                !pg_DoubleFromObj(PyTuple_GetItem(tup, 1), &y1) ||
+                !pg_DoubleFromObj(PyTuple_GetItem(arg, 0), &x2) ||
+                !pg_DoubleFromObj(PyTuple_GetItem(arg, 1), &y2)) {
+                PyErr_SetString(PyExc_TypeError,
+                                "Elements may only contain numbers");
+                return -1;
+            }
+            Py_DECREF(tup);
+
+            if (x1 == x2 && y1 == y2) {
+                return 1;
+            }
+        }
+        return 0;
+    }
+    else {
+        PyErr_SetString(PyExc_TypeError, "Expected a tuple");
+        return -1;
+    }
+}
```

Comment:
I don't like this either, as @Emc2356 suggested, this is unnecessary complexity, you can convert the input x and y values once and then compare them with any other double contained in the vertices:
See if this works

```C
static int
pg_polygon_contains_seq(pgPolygonObject *self, PyObject *arg)
{
    if (!PySequence_FAST_CHECK(arg)) {
        PyErr_SetString(PyExc_TypeError, "Expected a tuple");
        return -1;
    }

    pgPolygonBase *poly = &self->polygon;

    double x, y;
    if (!pg_TwoDoublesFromObj(arg, &x, &y)) {
        PyErr_SetString(PyExc_TypeError, "Expected a sequence of 2 numbers");
        return -1;
    }

    Py_ssize_t i;
    for (i = 0; i < poly->verts_num; i += 2) {
        if (poly->verts[i] == x && poly->verts[i + 1] == y) {
            return 1;
        }
    }

    return 0;
}
```

Reply:
i say we use the one that i sent on discord, it is better as it covers the sequence protocol and not just tuple and list